### ISSUE TITLE
Keyboard navigation with Up/Down arrows does not work in ListBox/TreeView when CheckBox is used in DataTemplate

### ISSUE THREAD

**Issue by wieslawsoltes** [AvaloniaUI/Avalonia#495]
``` XAML
<ListBox Name="layers" Items="{Binding Layers}" SelectedItem="{Binding CurrentLayer, Mode=TwoWay}" Classes="default" Grid.Row="1">
    <ListBox.KeyBindings>
        <KeyBinding Command="{Static Editor:Commands.AddLayerCommand}" CommandParameter="{Binding #layers.DataContext, Mode=OneWay}" Gesture="Ctrl+N"/>
        <KeyBinding Command="{Static Editor:Commands.RemoveLayerCommand}" CommandParameter="{Binding #layers.SelectedItem}" Gesture="Delete"/>
    </ListBox.KeyBindings>
    <ListBox.DataTemplates>
        <DataTemplate DataType="Project:XLayer">
            <Grid ColumnDefinitions="Auto,*">
                <Grid.ContextMenu>
                    <ContextMenu>
                        <MenuItem Header="_Delete" Command="{Static Editor:Commands.RemoveLayerCommand}" CommandParameter="{Binding Mode=OneWay}"/>
                    </ContextMenu>
                </Grid.ContextMenu>
                <CheckBox Content="" IsChecked="{Binding IsVisible, Mode=TwoWay}" Classes="default" Grid.Column="0"/>
                <TextBlock Text="{Binding Name}" Classes="default" Grid.Column="1"/>
            </Grid>
        </DataTemplate>
    </ListBox.DataTemplates>
</ListBox>
```

**Comment by grokys** [comment]
@wieslawsoltes I was unable to test this actual repro as it contains Core2D-specific stuff, but I tried creating a simple listbox like this:

```
      <ListBox Items="{Binding}" Width="250" Height="350">
        <ListBox.ItemTemplate>
          <DataTemplate>
            <CheckBox Content="{Binding}"/>
          </DataTemplate>        
        </ListBox.ItemTemplate>
      </ListBox>
```

And everything seemed to work. Closing this as it seems to be fixed. If you can provide a minimal failing repro I will repoen.
